guides/plugins/apps/content/cms/add-custom-cms-blocks.md

```diff
@@ -0,0 +1,161 @@
+# Add custom CMS blocks
+
+{% hint style="info" %}
+This functionality is available starting with Shopware 6.4.2.0.
+{% endhint %}
+
+Didn't get in touch with Shopware's Shopping Experiences \(CMS\) yet? Check out the concept behind it first:
+
+{% page-ref page="../../../../concepts/commerce/core/shopping-experiences-cms.md" %}
+
+If you're not yet familiar with custom CMS blocks you might want to head over to the section on [adding CMS blocks via plugin](../../../plugins/content/cms/add-cms-block.md) to get some basic understanding.
+
+{% page-ref page="../../../plugins/content/cms/add-cms-block.md" %}
+
+## Overview
+
+Adding custom CMS blocks from an app works a bit differently than [adding them from a plugin](../../../plugins/content/cms/add-cms-block.md).
+Custom CMS blocks are added by providing a `cms.xml` in the `Resources/` directory of your app.
+The basic directory structure looks as follows:
+
+```text
+├── Resources
+│   ├── cms
+│   │   └── blocks
+│   │       ├── my-first-block
+│   │       │   ├── preview.html
+│   │       │   └── styles.css
+│   │       └── my-second-block
+│   │           ├── preview.html
+│   │           └── styles.css
+│   └── cms.xml
+└── manifest.xml
+```
+
+Each CMS block defined within your `cms.xml` must have a directory matching the block's name in `Resources/cms/blocks/`.
+In those directories you shape your blocks by supplying a `preview.html` containing the template used for displaying a preview.
+Styling the preview is possible from the `styles.css`.
+
+{% hint type=info %}
+Due to technical limitations it's not possible to use templating engines \(like Twig\) for the preview template or preprocessors \(like Sass\) for styling.
+{% endhint %}
+
```

Comment:
We should tell the app manufacturers that they still have to add the storefront representation of their blocks to the `Resources/views/storefront/cms/block/` folder.

Reply:
Added section `## Storefront representation` with a short description and linking to the plugin example.